processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java

```diff
@@ -603,7 +603,7 @@ private void copyLocalFileToCarbonStore(String carbonStoreFilePath, String local
     short numberOfByteToStorLength = 2;
     // as length part is removed, so each no dictionary value index
     // needs to be reshuffled by 2 bytes
-    for (int i = 0; i < dataWriterVo.getNoDictionaryCount(); i++) {
+    for (int i = 0; i < dataWriterVo.getSegmentProperties().getNumberOfNoDictSortColumns(); i++) {
```

Comment:
Please create a local variable for `dataWriterVo.getSegmentProperties().getNumberOfNoDictSortColumns()`, it is better not put in loop